servers/quarkus-authn/src/main/java/org/projectnessie/server/authn/NessieAnonymousIdentityProvider.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2024 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.server.authn;
+
+import io.quarkus.security.identity.AuthenticationRequestContext;
+import io.quarkus.security.identity.IdentityProvider;
+import io.quarkus.security.identity.SecurityIdentity;
+import io.quarkus.security.identity.request.AnonymousAuthenticationRequest;
+import io.quarkus.security.runtime.AnonymousIdentityProvider;
+import io.smallrye.mutiny.Uni;
+import jakarta.enterprise.context.ApplicationScoped;
+
+@ApplicationScoped
+public class NessieAnonymousIdentityProvider
+    implements IdentityProvider<AnonymousAuthenticationRequest> {
+
+  private final AnonymousIdentityProvider anonymousIdentityProvider;
+
+  public NessieAnonymousIdentityProvider() {
+    this.anonymousIdentityProvider = new AnonymousIdentityProvider();
+  }
+
+  @Override
+  public Class<AnonymousAuthenticationRequest> getRequestType() {
+    return AnonymousAuthenticationRequest.class;
+  }
+
+  @Override
+  public Uni<SecurityIdentity> authenticate(
+      AnonymousAuthenticationRequest request, AuthenticationRequestContext context) {
+    return anonymousIdentityProvider.authenticate(request, context);
```

Comment:
nit: would it work if we just had a "provider" for this `AnonymousIdentityProvider` (as opposed to delegating to it)?